website/content/docs/v0.9/Guides/upgrading-talos.md

```diff
@@ -3,14 +3,54 @@ title: Upgrading Talos
 ---
 
 Talos upgrades are effected by an API call.
-The `talosctl` CLI utility will facilitate this, or you can use the automatic upgrade features provided by the [talos controller manager](https://github.com/talos-systems/talos-controller-manager).
+The `talosctl` CLI utility will facilitate this.
+<!-- , or you can use the automatic upgrade features provided by the [talos controller manager](https://github.com/talos-systems/talos-controller-manager) -->
 
 ## Video Walkthrough
 
 To see a live demo of this writeup, see the video below:
 
 <iframe width="560" height="315" src="https://www.youtube.com/embed/sw78qS8vBGc" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
 
+## Upgrading from Talos 0.8
+
+Talos 0.9 drops support for `bootkube` and self-hosted control plane.
+
+Please make sure Talos is upgraded to the latest minor release of 0.8 first (0.8.4 at the moment
+of this writing), then proceed with upgrading to the latest minor release of 0.9.
+
+### Before Upgrade to 0.9
+
+If cluster was bootstrapped on Talos version < 0.8.3, add checkpointer annotations to
+the `kube-scheduler` and `kube-controller-manager` daemonsets to improve resiliency of
+self-hosted control plane to reboots (this is critical for single control-plane node clusters):
+
+```bash
+$ kubectl -n kube-system patch daemonset kube-controller-manager --type json -p '[{"op": "add", "path":"/spec/template/metadata/annotations", "value": {"checkpointer.alpha.coreos.com/checkpoint": "true"}}]'
+daemonset.apps/kube-controller-manager patched
+$ kubectl -n kube-system patch daemonset kube-scheduler --type json -p '[{"op": "add", "path":"/spec/template/metadata/annotations", "value": {"checkpointer.alpha.coreos.com/checkpoint": "true"}}]'
+daemonset.apps/kube-scheduler patched
+```
+
+Talos 0.9 only supports Kubernetes versions 1.19.x and 1.20.x, if running 1.18.x, please upgrade Kubernetes before
+upgrading Talos.
+
+Make sure cluster is running latest minor release of Talos 0.8.
+
+Prepare by downloading `talosctl` binary for Talos release 0.9.x.
+
+### After Upgrade to 0.9
+
+After upgrade to 0.9, Talos will still run self-hosted control plane until [conversion process](../converting-control-plane/) is run.
```

Comment:
"After the upgrade to v0.9, Talos will still be running the self-hosted (bootkube) control plane until the [conversion process](../converting-control-plane/) is run."